forum/notifications/serializers.py

```diff
@@ -0,0 +1,38 @@
+from rest_framework import serializers
+from .models import NotificationType, NotificationPreference
+from django.contrib.auth.models import User
+
+
+class NotificationTypeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = NotificationType
+        fields = ['id', 'name', 'description']
+
+
+class NotificationPreferenceSerializer(serializers.ModelSerializer):
+    """
+    Serializer for reading user's notification preferences.
+    """
+    allowed_notification_types = serializers.SerializerMethodField()
+
```

Comment:
In NotificationPreferenceSerializer, it is better to make the user field read-only to avoid changing the user through requests:

user = serializers.ReadOnlyField(source='user.username')